numpy/core/_methods.py

```diff
@@ -230,3 +231,14 @@ def _ptp(a, axis=None, out=None, keepdims=False):
         umr_minimum(a, axis, None, None, keepdims),
         out
     )
+
+def _dump(self, file, protocol=2):
+    if hasattr(file, 'write'):
+        ctx = contextlib_nullcontext(file)
+    else:
+        ctx = open(os_fspath(file), "wb")
```

Comment:
Windows has readonly files, will this overwrite the file?

Reply:
No idea - but this is exactly what the pickle examples tell you to do when opening files.